plugins/lua/sort.lua

```diff
@@ -73,6 +75,28 @@ local function sort_by_migrant_wave_asc(unit_id_1, unit_id_2)
     return utils.compare(unit1.id, unit2.id)
 end
 
+local function sort_by_stress_desc(unit_id_1, unit_id_2)
+    if unit_id_1 == unit_id_2 then return 0 end
+    local unit1 = df.unit.find(unit_id_1)
+    local unit2 = df.unit.find(unit_id_2)
+    if not unit1 then return -1 end
+    if not unit2 then return 1 end
+    local happiness1 = unit1.status.current_soul.personality.stress
+    local happiness2 = unit2.status.current_soul.personality.stress
+    return utils.compare(happiness2, happiness1)
```

Comment:
if happiness1 == happiness2, then the ordering should fall back to sorting by name. otherwise the sort order is not stable for units with the same happiness, causing the list order to jump around if you click outside the assignment window and trigger a re-sort.